tests/test_treasury_module.py

```diff
@@ -94,6 +105,179 @@ def test_bundle_creation_with_riskpool_wallet_not_set(
+
+    bundle = riskpool.getBundle(0)
+    print(bundle)
+
+    (bundleId) = bundle[0]
+
+    # check bundle values with expectation
+    assert bundleId == 1
+    
+    # close bundle
+    riskpool.closeBundle(bundleId, {'from': capitalOwner})
+
+    # prepare allowance that is too small for bundle withdrawal
+    testCoin.approve(instance.getTreasury(), 0.9 * amount, {'from': capitalOwner})
+
+    # ensures that burning bundle fails during token withdrawal due to insufficient allowance
+    with brownie.reverts("ERC20: insufficient allowance"):
+        riskpool.burnBundle(
+                bundleId, 
+                {'from': capitalOwner})
+
+
+def test_payout_allowance_too_small(
+    instance: GifInstance,
+    owner: Account,
+    testCoin,
+    productOwner: Account,
+    oracleProvider: Account,
+    riskpoolKeeper: Account,
+    capitalOwner: Account,
+    customer: Account,
+):  
+    applicationFilter = bytes(0)
+
+    (gifProduct, gifRiskpool) = getProductAndRiskpool(
+        instance,
+        owner,
+        testCoin,
+        productOwner,
+        oracleProvider,
+        riskpoolKeeper,
+        capitalOwner,
+        True
+    )
+
+    # prepare product and riskpool
+    safetyFactor = 2
+    amount = 10000
+    testCoin.transfer(capitalOwner, safetyFactor * amount, {'from': owner})
+    testCoin.approve(instance.getTreasury(), amount, {'from': capitalOwner})
+    riskpool = gifRiskpool.getContract()
+
+    # fund bundle
+    riskpool.createBundle(
+                applicationFilter, 
+                amount, 
+                {'from': capitalOwner})
+
+    bundle = riskpool.getBundle(0)
+    print(bundle)
+
+    (bundleId) = bundle[0]
+
+    # check bundle values with expectation
+    assert bundleId == 1
+
+    # prepare prolicy application
+    premium = 100
+    sumInsured = 1000
+    product = gifProduct.getContract()
+    policyController = instance.getPolicy()
+
+    policyId = apply_for_policy(instance, owner, product, customer, testCoin, premium, sumInsured)
+
+    claimAmount = 800
+    instanceService = instance.getInstanceService()
+
+    # submit a claim for the policy
+    tx = product.submitClaimWithDeferredResponse(policyId, claimAmount, {'from': customer})
+    (claimId, requestId) = tx.return_value
+    claim = instanceService.getClaim(policyId, claimId).dict()
+    print(claim)
+
+    assert claim["state"] ==  0 # enum ClaimState {Applied, Confirmed, Declined, Closed}
+
+    # confirm the claim
+    product.confirmClaim(policyId, claimId, claimAmount, {'from': productOwner})
+
+    claim = instanceService.getClaim(policyId, claimId).dict()
+    assert claim["state"] ==  1 # enum ClaimState {Applied, Confirmed, Declined, Closed}
+
+    # check that it's possible to create payout for claim in confirmed state
+    payoutAmount = claimAmount
+    
+    # prepare allowance for payout that is too small
+    testCoin.approve(instance.getTreasury(), payoutAmount * 0.9, {'from': productOwner})

```

Comment:
the approval needs to come from the riskpoolWallet, not the productOwner

Reply:
done